### PR TITLE
Encoded data length issues

### PR DESCRIPTION
**Pull request by allenhsu**
Use `strlen()` instead of `length` of `NSString`

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCEncodingUtils.m [hunk 1]
```diff
@@ -199,7 +199,7 @@ + (NSString *)sanitizedStringFromString:(NSString *)dirtyString {
 
 + (NSData *)encodeDictionaryToJsonData:(NSDictionary *)dictionary {
     NSString *jsonString = [BNCEncodingUtils encodeDictionaryToJsonString:dictionary];
-    return [NSData dataWithBytes:[jsonString UTF8String] length:jsonString.length];
+    return [NSData dataWithBytes:jsonString.UTF8String length:strlen(jsonString.UTF8String)];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCEncodingUtils.m]: For style preference, could you use the other method you described in #157? Specifically, I prefer not using dot notation for method calls, and here you make the same method call (`UTF8String`) twice.

Comment (allenhsu) [inline Branch-SDK/Branch-SDK/BNCEncodingUtils.m]: OK, I'll do that. But in fact, `UTF8String` is a `property` of `NSString` pointing to a structure inside the string object, and there's no overhead calling twice.

**Comment by allenhsu** [comment]
@shortstuffsushi updated

File: Branch-SDK/Branch-SDK/BNCEncodingUtils.m (hunk 1 above)
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCEncodingUtils.m]: So it is, my mistake!

**Pull request merged**